src/rpc/blockchain.cpp

```diff
@@ -814,6 +814,32 @@ static bool GetUTXOStats(CCoinsView *view, CCoinsStats &stats)
     return true;
 }
 
+UniValue pruneblockchain(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() != 1)
+        throw runtime_error(
+            "pruneblockchain\n"
+            "\nArguments:\n"
+            "1. \"height\"       (int, required) The block height to prune up to.\n");
+
+    if (pruneMode != PRUNE_MANUAL) {
+        throw JSONRPCError(RPC_METHOD_NOT_FOUND, "Cannot prune via RPC unless in manual prune mode.");
+        return 0;
+    }
+
+    LOCK(cs_main);
+
+    int height = request.params[0].get_int();
+    if (height < 0) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Negative block height.");
+    } else if ((unsigned int) chainActive.Height() < Params().PruneAfterHeight()) {
+        throw JSONRPCError(RPC_INTERNAL_ERROR, "Blockchain is too short for pruning.");
+    }
```

Comment:
Do we need any foot-shooting checks: w.g. that the last 144 blocks are being retained to be robust against reorgs?